src/plopp/backends/matplotlib/canvas.py

```diff
@@ -159,10 +170,20 @@ def autoscale(self):
                         scale=self.yscale,
                     )
                 )
-                self._xmin = min(self._xmin, left.value)
-                self._xmax = max(self._xmax, right.value)
-                self._ymin = min(self._ymin, bottom.value)
-                self._ymax = max(self._ymax, top.value)
+                xmin = min(xmin, left.value)
+                xmax = max(xmax, right.value)
+                ymin = min(ymin, bottom.value)
+                ymax = max(ymax, top.value)
+        if self._autoscale == 'grow':
```

Comment:
Why does `autoscale` (which is for the colorbar) affect the behavior for X and Y limits?